uwrt_arm/uwrt_arm_controllers/uwrt_cartesian_controller/src/cartesian_controller.cpp

```diff
@@ -0,0 +1,222 @@
+  KDL::SegmentMap::iterator it;
+
+  for( it=segment_map.begin(); it != segment_map.end(); it++ ){
+    ROS_INFO_STREAM( "    "<<(*it).first);
+  }
+
+  ROS_INFO_STREAM("***Listing HW Interfaces: ***");    
+  // Get joint handles for all of the joints in the chain
+  for(std::vector<KDL::Segment>::const_iterator it = kdl_chain_.segments.begin(); it != kdl_chain_.segments.end(); ++it)
+  {
+    // WE DON'T WANT AN INTERFACE FOR THE FIXED JOINT
+    // if(it->getJoint().getName().compare("arm_shoulder_bicep_joint")) {
+    if(it->getJoint().getName() != "arm_shoulder_bicep_joint") {
+      joint_handles_.push_back(hw->getHandle(it->getJoint().getName()));
+      joint_names_.push_back(it->getJoint().getName());
+      ROS_INFO_STREAM("    "<<it->getJoint().getName());
+    }
+  }
+  ROS_INFO_STREAM("***KDL Chain has "<< kdl_chain_.getNrOfJoints() <<" joints ***");    
+
+  // KDL solvers
+  // NOLINTNEXTLINE(cppcoreguidelines-owning-memory)
+  cart_to_joint_solver_vel_.reset(new KDL::ChainIkSolverVel_pinv(kdl_chain_));
+
+  // Subscribe to twist msg 
+  arm_command_sub_ = nh.subscribe<uwrt_arm_msgs::UWRTArmTwist>("twist_arm_cmds",
+                                                                1,
+                                                                &UWRTCartesianController::armCommandCallback,
+                                                                this);
+
+  double dead_man_timeout;
+  // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+  nh.param<double>("dead_man_timeout", dead_man_timeout, 0.2);
+  dead_man_timeout_ = ros::Duration(dead_man_timeout);
+
+  return true;
+}
+
+
+ void UWRTCartesianController::armCommandCallback(const uwrt_arm_msgs::UWRTArmTwistConstPtr& arm_command)
+  {
+    qdot_cart_.vel(0) = arm_command->twist.linear.x;
+    qdot_cart_.vel(1) = arm_command->twist.linear.y;
+    qdot_cart_.vel(2) = arm_command->twist.linear.z;
+    qdot_cart_.rot(0) = arm_command->twist.angular.x;
+    qdot_cart_.rot(1) = arm_command->twist.angular.y;
+    qdot_cart_.rot(2) = arm_command->twist.angular.z;
+
+    last_msg_ = ros::Time::now();
+    got_msg_ = false;
+  }
+
+  // NOLINTNEXTLINE(misc-unused-parameters)
+  void UWRTCartesianController::starting(const ros::Time& time)
+  {
+    // set twists to zero
+    // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+    qdot_cart_[0] = 0.0;
+    // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+    qdot_cart_[1] = 0.0;
+    // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+    qdot_cart_[2] = 0.0;
+    // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+    qdot_cart_[3] = 0.0;
+    // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+    qdot_cart_[4] = 0.0;
+    // NOLINTNEXTLINE(readability-magic-numbers, cppcoreguidelines-avoid-magic-numbers)
+    qdot_cart_[5] = 0.0;
```

Comment:
```suggestion
    KDL::SetToZero(qdot_cart_);
```
This is better as it is clearer and avoids the lint warnings